addons/payment/models/payment_acquirer.py

```diff
@@ -170,30 +176,28 @@ def _compute_view_configuration_fields(self):
             'show_cancel_msg': True,
         })
 
-    #=== ONCHANGE METHODS ===#
-
-    @api.onchange('state')
-    def _onchange_state(self):
-        """ Only enable dashboard display for journals of enabled acquirers.
-
-        :return: None
-        """
+    @api.depends('payment_method_line_id')
+    def _compute_journal_id(self):
         for acquirer in self:
-            acquirer.journal_id.show_on_dashboard = acquirer.state == 'enabled'
-
-    @api.onchange('allow_tokenization')
-    def _onchange_allow_tokenization(self):
-        """ Add (remove) the electronic payment method for acquirers (not) allowing tokenization.
+            if acquirer.payment_method_line_id:
+                acquirer.journal_id = acquirer.payment_method_line_id.journal_id
+            else:
+                acquirer.journal_id = False
 
-        :return: None
-        """
-        electronic = self.env.ref('payment.account_payment_method_electronic_in')
+    def _inverse_journal_id(self):
         for acquirer in self:
-            if acquirer.allow_tokenization:
-                if electronic not in acquirer.journal_id.inbound_payment_method_ids:
-                    acquirer.journal_id.inbound_payment_method_ids = [(4, electronic.id)]
-            elif electronic in acquirer.journal_id.inbound_payment_method_ids:
-                acquirer.journal_id.inbound_payment_method_ids = [(3, electronic.id)]
+            if acquirer.journal_id:
+                if not acquirer.payment_method_line_id:
+                    acquirer.payment_method_line_id = self.env['account.payment.method.line'].create(
+                        acquirer._get_default_payment_method_info()
+                    )
+                else:
+                    acquirer.payment_method_line_id.journal_id = acquirer.journal_id.id
+            elif acquirer.payment_method_line_id:
+                acquirer.payment_method_line_id.unlink()
```

Comment:
I think we should avoid this behavior if the acquirer is in `test` or `enabled`. Then, you could rewrite it with:
```
payment_method = self.env['account.payment.method.line'].search([('code', '=', acquirer.provider)])
payment_method.journal_id = acquirer.journal_id
```